test/basic.test.ts

```diff
@@ -482,3 +484,181 @@ describe('app config', () => {
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/BreaksServer.9d9aeeb9.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/injectAuth.2083679e.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/override.af1cd5ce.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/foo.72febae8.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/error-component.eceba4d2.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/PascalCase.87deb145.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/custom.daf0d259.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/override.62df4802.js",
+              "rel": "prefetch",
+            },
+          ],
+          [
+            "link",
+            {
+              "as": "script",
+              "href": "/_nuxt/default.b7809d21.js",
+              "rel": "prefetch",
+            },
+          ],
+        ],
+      }
+    `)
+    expect(result.state).toMatchInlineSnapshot('{}')
+    expect(Object.keys(result)).toMatchInlineSnapshot(`
+      [
+        "html",
+        "state",
+        "tags",
+      ]
+    `)
+  })
+  it('renders pure components', async () => {
+    const result: NuxtIslandResponse = await $fetch(withQuery('/__nuxt_island/PureComponent', {
+      props: JSON.stringify({
+        bool: false,
+        number: 3487,
+        str: 'something',
+        obj: { foo: 42, bar: false, me: 'hi' }
+      })
+    }))
+
+    expect(result.html.replace(/data-v-\w+|"|<!--.*-->/g, '')).toMatchInlineSnapshot(`
+      "<div > Was router enabled: true <br > Props: <pre >{
+        number: 3487,
+        str: something,
+        obj: {
+          foo: 42,
+          bar: false,
+          me: hi
+        },
+        bool: false
+      }</pre></div>"
+    `)
+    expect(result.state).toMatchInlineSnapshot(`
+      {
+        "$shasRouter": true,
+      }
+    `)
+
+    // TODO: Test response differes!
+    // expect(result.html.head.join(' ')).toMatch(/PureComponent[^"']*.css/)
```

Comment:
```suggestion
    expect(result.html.head.join(' ')).toMatch(/PureComponent[^"']*.css/)
```

- [ ] resolve before merging

See https://github.com/nuxt/framework/pull/5689#discussion_r949365675. I suspect this is because (as commented there) the CSS is not being generated in client side build.

Reply:
https://github.com/nuxt/framework/pull/5689#discussion_r949578267 i'm aware it is broken but main issue is that styles are not currently properly inlined from server bundle while they should. It is not unique issue to islands.

Reply:
I believe this is the intended behaviour from vite-plugin-vue, and not broken.

It would certainly be a nice enhancement to have the option to render inlined styles, of course, albeit at the cost of an increased server bundle size.

Reply:
Let's track in #6755 but if it is actually behavior/limitation of `vite-plugin-vue` for server build, it means it is not ready for server-only components because transform plugin is essentially making partial use of client build to make styles working.